### DISCUSSION TITLE
React project Tailwind CSS overwriting component library tailwind css

### DISCUSSION THREAD

**Discussion by husnainali92** [tailwindlabs/tailwindcss#10849] [Help]
I created a component library using React-aria and TailwindCSS. I used rollup for building the library.
At library level, everything works fine, but when I import components from my library to my project,
Library styles are being overwritten by the project TailwindCSS (index.css) file.

![style overwritten](https://user-images.githubusercontent.com/101558118/227155272-f0345f24-ed69-4110-a8da-6d4701104327.png)

How can I prevent my project TailwindCSS from overwriting the library's tailwind styles?

This is the project index.css
![image](https://user-images.githubusercontent.com/101558118/227158081-af0b8abb-836c-489a-9b8d-f3c11fb44d8f.png)

**Comment by wongjn** [comment]
If the library is using similar Tailwind configuration, you could look at generating the Tailwind CSS for the library in your own consumer project and avoid importing the library's Tailwind CSS.

**Comment by farshid1** [comment]
I'm having a similar problem. Is there a native way of achieving this in tailwindcss? I feel like this is a common use case to build component libraries using tailwindcss and used them in apps that use tailwindcss as well.
